app/ts/components/TransferButton.tsx

```diff
@@ -27,17 +27,15 @@ export const TransferButton = () => {
 
 const ConnectOrTransferButton = () => {
 	const { value: query, waitFor } = useAsyncState<EthereumAddress>()
-	const { browserProvider } = useWallet()
-	const { account } = useAccount()
-	const { notify } = useNotice()
+	const { browserProvider } = useEthereumProvider()
+	const { account } = useWallet()
 
 	const connect = () => {
-		if (!browserProvider) {
-			notify({ message: 'No compatible web3 wallet detected.', title: 'Failed to connect' })
-			return
-		}
 		waitFor(async () => {
-			const signer = await browserProvider.getSigner()
+			if (!browserProvider.value) {
+				throw makeError('No compatible web3 wallet detected.', 'UNKNOWN_ERROR', { error: { code: 4900 } })
```

Comment:
this error message and code repeats, we could make it a constant or a function

Reply:
True, created an issue for this #212